include/BellareMicaliOTProtocol.hh

```diff
@@ -0,0 +1,54 @@
+#ifndef BELLARE_MICALI_OT_PROTOCOL_HH
+#define BELLARE_MICALI_OT_PROTOCOL_HH
+
+#include "Protocol.hh"
+#include "QuadraticResidueGroup.hh"
+
+// This Oblivious Transfer (OT) protocol was
+// first introduced by Bellare and Micali,
+// and later amended by Naor and Pinkas
+// in the following paper:
+// https://dl.acm.org/doi/10.5555/365411.365502
+
+enum class BellareMicaliOTStage {
+  Init,
+  SendPublicKey,
+  SendEncryptedMessages,
+  Done,
+};
+
+class BellareMicaliOTSender {
```

Comment:
Sender and Chooser objects store data for the sender and chooser party, respectively. Only one of them is 'instantiated' in an instance of the protocol.